sdks/java/io/snowflake/src/test/java/org/apache/beam/sdk/io/snowflake/test/SnowflakeIOIT.java

```diff
@@ -0,0 +1,174 @@
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.coders.SerializableCoder;
+import org.apache.beam.sdk.io.FileSystems;
+import org.apache.beam.sdk.io.GenerateSequence;
+import org.apache.beam.sdk.io.common.HashingFn;
+import org.apache.beam.sdk.io.common.TestRow;
+import org.apache.beam.sdk.io.fs.MatchResult;
+import org.apache.beam.sdk.io.fs.MoveOptions;
+import org.apache.beam.sdk.io.fs.ResourceId;
+import org.apache.beam.sdk.io.snowflake.SnowflakeIO;
+import org.apache.beam.sdk.io.snowflake.credentials.SnowflakeCredentialsFactory;
+import org.apache.beam.sdk.io.snowflake.enums.WriteDisposition;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PCollection;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * A test of {@link org.apache.beam.sdk.io.snowflake.SnowflakeIO} on an independent Snowflake
+ * instance.
+ *
+ * <p>This test requires a running instance of Snowflake, configured for your GCP account. Pass in
+ * connection information using PipelineOptions:
+ *
+ * <pre>
+ * ./gradlew -p sdks/java/io/snowflake integrationTest -DintegrationTestPipelineOptions='[
+ * "--serverName=<YOUR SNOWFLAKE SERVER NAME>",
+ * "--username=<USERNAME>",
+ * "--password=<PASSWORD>",
+ * "--database=<DATABASE NAME>",
+ * "--schema=<SCHEMA NAME>",
+ * "--stagingBucketName=gs://<GCS BUCKET NAME>",
+ * "--storageIntegrationName=<STORAGE INTEGRATION NAME>",
+ * "--numberOfRecords=<1000, 100000, 600000, 5000000>",
+ * "--runner=DataflowRunner",
+ * "--region=<GCP REGION FOR DATAFLOW RUNNER>",
+ * "--project=<GCP_PROJECT>"]'
+ * --tests org.apache.beam.sdk.io.snowflake.test.SnowflakeIOIT
+ * -DintegrationTestRunner=dataflow
+ * </pre>
+ */
+public class SnowflakeIOIT {
+  private static final String tableName = "IOIT";
+
+  private static SnowflakeIO.DataSourceConfiguration dataSourceConfiguration;
+  private static int numberOfRecords;
+  private static String stagingBucketName;
+  private static String storageIntegrationName;
+
+  @Rule public TestPipeline pipelineWrite = TestPipeline.create();
+  @Rule public TestPipeline pipelineRead = TestPipeline.create();
+
+  @BeforeClass
+  public static void setup() throws SQLException {
+    SnowflakeIOITPipelineOptions options =
+        readIOTestPipelineOptions(SnowflakeIOITPipelineOptions.class);
+
+    numberOfRecords = options.getNumberOfRecords();
+    stagingBucketName = options.getStagingBucketName();
+    storageIntegrationName = options.getStorageIntegrationName();
+
+    dataSourceConfiguration =
+        SnowflakeIO.DataSourceConfiguration.create(SnowflakeCredentialsFactory.of(options))
+            .withDatabase(options.getDatabase())
+            .withRole(options.getRole())
```

Comment:
In example of running test there is no role or warehouse parameter.

Reply:
👌